tests/test_atoms_state.py

```diff
@@ -0,0 +1,350 @@
+            (OrbitalsState(l_quantum_number=1, ml_quantum_number=-1), 2, 0.5, 1.5),
+            (None, None, 0.5, None),
+        ],
+    )
+    def test_occupation(self, orbital_ref, degeneracy, n_excited_electrons, occupation):
+        """
+        Test the occupation of a core hole for a given set of orbital reference and degeneracy.
+        """
+        core_hole = CoreHole()
+        core_hole.orbital_ref = orbital_ref
+        if orbital_ref is not None:
+            assert orbital_ref.resolve_degeneracy() == degeneracy
+        core_hole.n_excited_electrons = n_excited_electrons
+        resolved_occupation = core_hole.resolve_occupation(logger)
+        if resolved_occupation is not None:
+            assert np.isclose(resolved_occupation, occupation)
+        else:
+            assert resolved_occupation == occupation
+
+    @pytest.mark.parametrize(
+        'orbital_ref, n_excited_electrons, dscf_state, results',
+        [
+            (OrbitalsState(l_quantum_number=1), -0.5, None, (-0.5, None, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, None, (0.5, None, 5.5)),
+            (
+                OrbitalsState(l_quantum_number=1, ml_quantum_number=-1),
+                0.5,
+                None,
+                (0.5, None, 1.5),
+            ),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'initial', (None, 1, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'final', (0.5, None, 5.5)),
+            (None, 0.5, None, (0.5, None, None)),
+        ],
+    )
+    def test_normalize(self, orbital_ref, n_excited_electrons, dscf_state, results):
+        """
+        Test the normalization of the `CoreHole`. Inputs are defined as the quantities of the `CoreHole` section.
+        """
+        core_hole = CoreHole()
+        core_hole.orbital_ref = orbital_ref
+        core_hole.n_excited_electrons = n_excited_electrons
+        core_hole.dscf_state = dscf_state
+
+        core_hole.normalize(None, logger)
+
+        assert core_hole.n_excited_electrons == results[0]
+        if core_hole.orbital_ref:
+            assert core_hole.orbital_ref.degeneracy == results[1]
+            assert core_hole.orbital_ref.occupation == results[2]
+
+
+class TestHubbardInteractions:
+    """
+    Test the `HubbardInteractions` class defined in atoms_state.py.
+    """
+
+    @staticmethod
+    def add_u_j(hubbard_interactions, u, j) -> None:
+        """Adds `u_interaction` and `j_local_exchange_interaction` (in eV) to the `HubbardInteractions` section."""
+        if u is not None:
+            hubbard_interactions.u_interaction = u * ureg('eV')
+        if j is not None:
+            hubbard_interactions.j_local_exchange_interaction = j * ureg('eV')
+
+    @pytest.mark.parametrize(
+        'slater_integrals, results',
+        [
+            ([3.0, 2.0, 1.0], (0.1429146, -0.0357286, 0.0893216)),
+            (None, (None, None, None)),
+            ([3.0, 2.0, 1.0, 0.5], (None, None, None)),
```

Comment:
What does this last example test? An incorrect no. passed integrals by whom?
I'd rather see more integrals tested..?